File: apps/DigitalSkyServer/Domain/Message.cs
namespace DigitalSky.Domain
{
    public class Message
    {
        public string Id{ get; set; } = "";
        public string ClientId{ get; set; } = "";
        public string Target{ get; set; } = "*";
        public string Type { get; set; } = "";
        public string Content { get; set; } = "";
    }
}

File: packages/digital_sky_common/lib/src/domain/message.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'message.freezed.dart';
part 'message.g.dart';

enum MessageType {
  ping,
  masterJoin,
  playerJoin,
  playerJoinFinished,
  playerList,
  playerLeft,
}

@freezed
abstract class Message with _$Message {
  const factory Message({
    @Default(null) String? id,
    @Default(null) String? clientId,
    @Default("*") String target,
    required MessageType type,
    required String content,
  }) = _Message;

  factory Message.fromJson(Map<String, dynamic> json) => _$MessageFromJson(json);
}
